### ISSUE TITLE
Titan-core fails when an error is encountered (e.g. feature not available in this edition/license)

### ISSUE THREAD

**Issue by dsmdavid** [Titan-Systems/titan#90]
Thanks for your work here!

I'm testing Titan in an Snowflake "Trial Account". 
When executing: `python -m titan export --all --out=test.yml`, **_titan_** fails not too gracefully due to some features that are not available (e.g. compute pools)

```
─❯ python -m titan export --all --out=test.yml
[OWNER:ACCOUNTADMIN] > SHOW ALERTS IN ACCOUNT    (0 rows, 0.34s)
[OWNER:ACCOUNTADMIN] > SHOW API INTEGRATIONS    (0 rows, 0.08s)
[OWNER:ACCOUNTADMIN] > SHOW AUTHENTICATION POLICIES IN ACCOUNT    (0 rows, 0.34s)
[OWNER:ACCOUNTADMIN] > SHOW CATALOG INTEGRATIONS    (0 rows, 0.07s)
[OWNER:ACCOUNTADMIN] > SHOW COMPUTE POOLS    (err 395054, 0.22s)
Traceback (most recent call last):
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/titan/client.py", line 77, in execute
    result = cur.execute(sql_text).fetchall()
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/snowflake/connector/cursor.py", line 1080, in execute
    Error.errorhandler_wrapper(self.connection, self, error_class, errvalue)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/snowflake/connector/errors.py", line 290, in errorhandler_wrapper
    handed_over = Error.hand_to_other_handler(
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/snowflake/connector/errors.py", line 345, in hand_to_other_handler
    cursor.errorhandler(connection, cursor, error_class, error_value)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/snowflake/connector/errors.py", line 221, in default_errorhandler
    raise error_class(
snowflake.connector.errors.ProgrammingError: 395054 (22023): Snowpark Container Services feature COMPUTE POOL not available for trial accounts

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.9/3.9.18_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/local/Cellar/python@3.9/3.9.18_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/titan/__main__.py", line 5, in <module>
    titan_cli()
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/click/core.py", line 1688, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/titan/cli.py", line 130, in export
    resource_config = export_resources(exclude=_parse_resources(exclude))
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/titan/operations/export.py", line 23, in export_resources
    config.update(export_resource(session, resource_type))
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/titan/operations/export.py", line 31, in export_resource
    resource_names = list_resource(session, resource_label)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/titan/data_provider.py", line 1858, in list_resource
    return getattr(__this__, f"list_{pluralize(resource_label)}")(session)
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/titan/data_provider.py", line 1903, in list_compute_pools
    show_result = execute(session, "SHOW COMPUTE POOLS")
  File "/PATH_TO_PROJECT/titan/.env/lib/python3.9/site-packages/titan/client.py", line 87, in execute
    raise ProgrammingError(f"failed to execute sql, [{sql_text}]", errno=err.errno) from err
snowflake.connector.errors.ProgrammingError: 395054: failed to execute sql, [SHOW COMPUTE POOLS]
```

I found something similar with a database created on a share (which is fixed in this newest version).  
In my mind, it could be useful to have a flag that could be set to enable it to continue after errors (maybe just summarizing the errors at the end). At least for the "export" part, I understand for the "apply" we may not be so lenient with errors.

**Comment by teej** [comment]
Snowflake annoyingly doesn't provide a direct way to check account edition or trial status. This one is tricky because COMPUTE POOLS are available in standard edition just not during trial mode. And I can't automate trial mode checks without manually creating throwaway emails and accounts.

There's an undocumented system function `SYSTEM$GET_TRIAL_USAGE()` but I just tried it and it seems to fail.

I can write a quick fix for now.

**Comment by dsmdavid** [comment]
That's a good one. 
In the trial account the function `select SYSTEM$GET_TRIAL_USAGE();` works correctly and returns:
```
{"organization_id":"1028_f3889eb6-64c0-4ac9-963a-2069224233c9","remaining_free_usage_amount":39999,"remaining_free_usage_currency":"USD","remaining_free_usage_date":1721952000,"total_allocated_free_usage_amount":40000,"total_allocated_free_usage_currency":"USD","contract_end_date":1724630400,"updated_on":1722028506}
```
but in a regular account (running as accountadmin) I get an error:
```
Object '1003_73565840-b7c6-4773-8f46-734606322507' does not exist or not authorized.
```

**Comment by titan-teej** [comment]
Fixed in [v0.8.8](https://github.com/Titan-Systems/titan/releases/tag/v0.8.8). I didn't run into any other resource types that throw an error in trial accounts.
